feat(select): add KolSelectWc component and update component names

yurtsever opened the pull request:
## Summary
Refactors `kol-select` to delegate rendering and logic to a new `kol-select-wc` web component, and updates `kol-pagination` to use `kol-select-wc` for the page-size selector.

## Changes
- Extract `kol-select` logic into new `KolSelectWc` (`kol-select-wc`) component
- Simplify `kol-select` shadow component to a thin wrapper around `kol-select-wc`
- Update `kol-pagination` to use `KolSelectWcTag` for page size selection
- Add `.page-size` CSS class and import select styles into pagination stylesheet
- Update test snapshots to reflect new component structure
- Fix minor apostrophe in `kol-select` README

---

<details>
<summary>🇩🇪 Deutsche Zusammenfassung</summary>

## Zusammenfassung
Refaktoriert `kol-select`, indem Rendering und Logik an eine neue `kol-select-wc`-Web-Komponente delegiert werden, und aktualisiert `kol-pagination` auf die Nutzung von `kol-select-wc` für die Seitengrößen-Auswahl.

## Änderungen
- `kol-select`-Logik in neue `KolSelectWc`-Komponente (`kol-select-wc`) extrahiert
- `kol-select`-Shadow-Komponente als einfache Wrapper-Komponente vereinfacht
- `kol-pagination` nutzt jetzt `KolSelectWcTag` für Seitengrößen-Auswahl
- `.page-size`-CSS-Klasse und Select-Styles in Pagination-Stylesheet hinzugefügt
- Test-Snapshots an neue Komponentenstruktur angepasst
- Kleiner Apostroph-Fix in `kol-select` README
</details>

deleonio commented on packages/components/src/components/pagination/shadow.tsx:
```diff
@@ -165,16 +165,20 @@ export class KolPagination implements PaginationAPI {
-						_value={[this.state._pageSize]}
-					></KolSelectTag>
+					<div class="page-size">
+						<label htmlFor={`pagination-size-${this.nonce}`}>{translate('kol-page-per-site')}</label>
```
Das Label ist Teil der Select-Komponent.

```suggestion
```

deleonio commented on packages/components/src/components/pagination/shadow.tsx:
```diff
@@ -165,16 +165,20 @@ export class KolPagination implements PaginationAPI {
+					<div class="page-size">
+						<label htmlFor={`pagination-size-${this.nonce}`}>{translate('kol-page-per-site')}</label>
+
+						<KolSelectTag
```
Hier müssen wir `KolSelectWcTag` verwenden und das Select-Styling in die Pagination importieren.

```suggestion
<KolSelectWcTag
	_label={translate('kol-entries-per-site')}
	_options={this.state._pageSizeOptions}
	_on={{
		onChange: this.onChangePageSize,
	}}
	_value={[this.state._pageSize]}
/>
```

deleonio replied:
@copilot refactor suggestion

deleonio committed:
feat(select): add KolSelectWc component and update component names

- Introduced a new web component `KolSelectWc` for enhanced select functionality.
- Added `KolSelectWcTag` to component names for consistency.
- Updated localization files to simplify page entry messages in German and English.

Refs: #8653

deleonio committed:
Update all snapshots$
$
Refs: #8653

deleonio committed:
fix(select): update getValue method to retrieve value from selectWcRef

Refs: #8653

BF150 committed:
label to the left$
$
Refs: #8653

BF150 committed:
Update all snapshots$
$
Refs: #8653

deleonio commented on packages/test-tag-name-transformer/snapshots/theme-default/snapshot-for-pagination-basic-firefox-linux.png:
Das Dropdown-Select in der Pagination soll den Style von einem Select haben. Nur dass das Label eben links statt darüber ist.

BF150 replied:
Da hatte mich das Bild von Christian im Ticket verwirrt, dachte das soll so klein sein. Jetzt ist es in Kolibri Optik. 

Was hat es mit den [bwst Snapshots](https://github.com/public-ui/kolibri/pull/8882/files#diff-5192cacbd90d28c457b54a89ffc6fb22d95609c70f731ec2fe1ec9c397092366) auf sich? Soll ich da den Style auch noch anpassen?

deleonio replied:
Machen wird dann intern

BF150 committed:
styling as normal kol select$
$
Refs: #8653

BF150 committed:
Update all snapshots$
$
Refs: #8653

BF150 committed:
linter$
$
Refs: #8653

deleonio reviewed (approved):
❤️
